File: bindings/lua/Polycode/RenderTargetBinding.lua
class "RenderTargetBinding"

RenderTargetBinding.MODE_IN = 0
RenderTargetBinding.MODE_OUT = 1
RenderTargetBinding.MODE_COLOR = 2
RenderTargetBinding.MODE_DEPTH = 3

function RenderTargetBinding:__getvar(name)
	if name == "id" then
		return Polycode.RenderTargetBinding_get_id(self.__ptr)
	elseif name == "name" then
		return Polycode.RenderTargetBinding_get_name(self.__ptr)
	elseif name == "mode" then
		return Polycode.RenderTargetBinding_get_mode(self.__ptr)
	end
end

function RenderTargetBinding:__setvar(name,value)
	if name == "id" then
		Polycode.RenderTargetBinding_set_id(self.__ptr, value)
		return true
	elseif name == "name" then
		Polycode.RenderTargetBinding_set_name(self.__ptr, value)
		return true
	elseif name == "mode" then
		Polycode.RenderTargetBinding_set_mode(self.__ptr, value)
		return true
	end
	return false
end
function RenderTargetBinding:__delete()
	if self then Polycode.delete_RenderTargetBinding(self.__ptr) end
end

File: bindings/javascript/Polycode/RenderTargetBinding.js
function RenderTargetBinding() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.RenderTargetBinding()
	}
	Object.defineProperties(this, {
		'id': { enumerable: true, configurable: true, get: RenderTargetBinding.prototype.__get_id, set: RenderTargetBinding.prototype.__set_id},
		'name': { enumerable: true, configurable: true, get: RenderTargetBinding.prototype.__get_name, set: RenderTargetBinding.prototype.__set_name},
		'mode': { enumerable: true, configurable: true, get: RenderTargetBinding.prototype.__get_mode, set: RenderTargetBinding.prototype.__set_mode}
	})
}

RenderTargetBinding.MODE_IN = 0
RenderTargetBinding.MODE_OUT = 1
RenderTargetBinding.MODE_COLOR = 2
RenderTargetBinding.MODE_DEPTH = 3

RenderTargetBinding.prototype.__get_id = function() {
	return Polycode.RenderTargetBinding__get_id(this.__ptr)
}

RenderTargetBinding.prototype.__set_id = function(val) {
	Polycode.RenderTargetBinding__set_id(this.__ptr, val)
}

RenderTargetBinding.prototype.__get_name = function() {
	return Polycode.RenderTargetBinding__get_name(this.__ptr)
}

RenderTargetBinding.prototype.__set_name = function(val) {
	Polycode.RenderTargetBinding__set_name(this.__ptr, val)
}

RenderTargetBinding.prototype.__get_mode = function() {
	return Polycode.RenderTargetBinding__get_mode(this.__ptr)
}

RenderTargetBinding.prototype.__set_mode = function(val) {
	Polycode.RenderTargetBinding__set_mode(this.__ptr, val)
}

Duktape.fin(RenderTargetBinding.prototype, function (x) {
	if (x === RenderTargetBinding.prototype) {
		return;
	}
	Polycode.RenderTargetBinding__delete(x.__ptr)
})
